### PR TITLE
Convert gh-search-input to ember-power-select component

### PR DESCRIPTION
**Pull request by acburdine**
refs #6458

TODO:
- [x] get search (kind-of) working
- [x] fix styles
- [x] add tests (sort of)
- [x] :100:

### CODE REVIEW

**Comment by kevinansfield** [comment]
I've been playing with this and a reasonably large dataset (~270 posts, ~50 tags) and it would seem that performance has decreased compared to the selectize-based input. It's usable but I think it could be improved by debouncing the keyboard input.

Another minor thing - is is possible to show a "Loading..." message instead of "No results found" immediately after typing before the result set has finished being fetched?

**Comment by acburdine** [comment]
@kevinansfield was having some really odd issues when attempting to do debouncing, will explain later.

**Comment by acburdine** [comment]
screencap of debouncing issues: http://recordit.co/WRYqpjVDXb

**Comment by kevinansfield** [comment]
Hmm, is the behaviour any better if you debounce in the search function rather than within the keydown handler? http://www.ember-power-select.com/cookbook/debounce-searches. There's also an interesting `ember-concurrency` example: http://ember-concurrency.com/#/docs/examples/autocomplete.

I think I'd like to start replacing our custom timeouts and polling with ember-concurrency, it should help testing in that the `timeout` function can be wrapped in a service and then easily stubbed in tests, e.g.: https://gist.github.com/achambers/c2c762717965c886ebf8695fb80df27a

File: core/client/app/templates/components/gh-search-input.hbs [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-    on-init="onInit"
-    on-focus="onFocus"
-    on-blur="onBlur"
-    update-filter="onType"
-    select-item="openSelected"}}
-<button class="gh-nav-search-button" {{action "focusInput"}}><i class="icon-search"></i><span class="sr-only">Search</span></button>
+    onopen=(action "onFocus")
+    onclose=(action "onBlur")
+    searchEnabled=false
+    loadingMessage=false
```
Comment (kevinansfield) [inline core/client/app/templates/components/gh-search-input.hbs]: I think this line can be removed as it's duplicated below?

File: core/client/app/components/gh-search-input.js [hunk 2]
```diff
@@ -127,7 +168,7 @@ export default Component.extend({
 
     actions: {
         openSelected(selected) {
-            let transition = null;
+            console.log(selected);
```
Comment (kevinansfield) [inline core/client/app/components/gh-search-input.js]: :trollface:

File: core/client/app/components/gh-search-input.js [hunk 3]
```diff
@@ -135,60 +176,32 @@ export default Component.extend({
...
-
-            if (!selectize.hasOptions) {
-                selectize.open();
-                selectize.$empty_results_container.show();
-            } else {
-                selectize.$empty_results_container.hide();
-            }
+        search(term) {
+            return new RSVP.Promise((resolve, reject) => {
+                run.debounce(this, this._performSearch, term, resolve, reject, 600);
```
Comment (kevinansfield) [inline core/client/app/components/gh-search-input.js]: After playing with it for a while I would suggest lowering the debounce period, `200ms` feels a lot more responsive even with a large dataset.

**Comment by kevinansfield** [comment]
:+1: LGTM, only a couple of very minor things that I've commented on directly :smile: 

One more thing, is it easy to display the `Loading` message immediately when starting with a blank input? If the results box is already present then the loading message is displayed as soon as you start typing but with a blank input/no results box there is nothing displayed until the debounce/search has finished - having something displayed immediately would make it feel more responsive even though you're not actually getting the results any quicker.

**Comment by acburdine** [comment]
@kevinansfield This is rebased and should be good to go 👍

**Comment by kevinansfield** [comment]
I think we still have that duplicated `loadingMessage` component attribute.

Did you have any thoughts on my query about the loading message above? No worries if it's difficult to do, it can be a separate PR, although if it's a simple thing it would be good to get it in alongside.

**Comment by acburdine** [comment]
@kevinansfield should be good now, I wasn't exactly sure how to do it at first but I think I figured it out. It should show the loading message when typing now.

**Pull request merged**